thewalrus/tests/test_operations.py

```diff
@@ -0,0 +1,166 @@
+            [0.6263739 + 0.09615331j, -0.22788717 + 0.13121343j, 0.36548296 - 0.0200537j, -0.20708137 + 0.14004403j, 0.25645667 - 0.06275564j],
+            [0.5425389 + 0.14442404j, 0.19268911 + 0.15615312j, 0.11497303 + 0.13744549j, 0.21448948 + 0.08109308j, -0.03652914 + 0.15069359j],
+            [-0.00915607 + 0.07475267j, 0.48081922 + 0.10576742j, -0.00961086 + 0.20535144j, 0.33089303 + 0.09864247j, 0.02555522 + 0.19950786j],
+            [-0.34614367 - 0.05229875j, 0.11543956 + 0.01112537j, 0.16672961 + 0.07439407j, 0.02323121 + 0.15103267j, 0.27233637 + 0.08297028j],
+            [-0.14390852 - 0.08884069j, -0.37898007 - 0.07630228j, 0.12911863 - 0.08963054j, -0.12164023 + 0.04431394j, 0.1141808 + 0.01581529j],
+        ]
+    )
+    T = operations.fock_tensor(S, alphas, cutoff)
+    assert np.allclose(T, expected, atol=tol, rtol=0)
+
+
+def test_hong_ou_mandel_interference(tol):
+    """Tests the properties of a 50-50 beamsplitter"""
+    cutoff = 5
+    nmodes = 2
+    U = np.sqrt(0.5) * np.array([[1, 1j], [1j, 1]])
+    alphas = np.zeros([nmodes])
+    S = interferometer(U)
+    T = operations.fock_tensor(S, alphas, cutoff)
+
+    items = []
+    for i in range(cutoff):
+        for j in range(i + 1):
+            items.append([i - j, j])
+
+    mat = np.empty([len(items), len(items)], dtype=np.complex128)
+    for i, indi in enumerate(items):
+        for j, indj in enumerate(items):
+            mat[i, j] = T[tuple(indj + indi)]
+            if sum(indj) != sum(indi):
+                assert np.allclose(T[tuple(indj + indi)], 0)
+                # Checking that different excitation manifolds are not mixed
+            if indj == [1, 1] and indi == [1, 1]:
+                assert np.allclose(T[tuple(indj + indi)], 0)
+                # This is Hong-Ou-Mandel interference
+            if indj == [0, 0] and indi == [0, 0]:
+                assert np.allclose(T[tuple(indj + indi)], 1.0)
+                # Checking that the vacuum-vacuum amplitude is 1
+    assert np.allclose(mat[1 : 1 + nmodes, 1 : 1 + nmodes], U, atol=tol, rtol=0)
+
+
+
+def test_single_excitation_manifold_unitary(tol):
+    """Tests the properties of a 3 mode interferometer"""
+    cutoff = 5
+    nmodes = 3
+    U = np.array(
+        [
+            [0.25962161 - 0.08744841j, -0.63742098 + 0.55640489j, -0.35243833 + 0.29128117j],
+            [0.73811606 - 0.32606317j, 0.33062786 - 0.32070986j, -0.28449121 + 0.23614116j],
+            [-0.46985047 - 0.23034198j, 0.26455752 + 0.04413416j, -0.79944267 - 0.12302853j],
+        ]
+    )
+    alphas = np.zeros([nmodes])
+    S = interferometer(U)
+    T = operations.fock_tensor(S, alphas, cutoff)
+
+    items = [[0, 0, 0], [1, 0, 0], [0, 1, 0], [0, 0, 1]]
+    mat = np.zeros([len(items), len(items)], dtype=np.complex128)
+    for i, indi in enumerate(items):
+        for j, indj in enumerate(items):
+            mat[i, j] = T[tuple(indi + indj)]
+    np.allclose(U, mat[1 : 1 + nmodes, 1 : 1 + nmodes], atol=tol, rtol=0)
```

Comment:
These two tests are a bit hard to follow, so might benefit from commenting :)

But great tests!